tools/bpf2c/bpf_code_generator.cpp

```diff
@@ -557,17 +551,17 @@ bpf_code_generator::encode_instructions(const bpf_code_generator::unsafe_string&
                 output.lines.push_back(format_string("%s = -(int64_t)%s;", destination, destination));
                 break;
             case AluOperations::Mod:
-                if (inst.opcode & EBPF_SRC_REG) {
-                    output.lines.push_back(check_div_by_zero);
-                } else if (inst.imm == 0) {
-                    throw bpf_code_generator_exception(
-                        "invalid instruction - constant division by zero", output.instruction_offset);
-                }
                 if (is64bit)
-                    output.lines.push_back(format_string("%s %%= %s;", destination, source));
+                    output.lines.push_back(format_string(
+                        "%s = %s ? (%s %% %s): %s ;", destination, source, destination, source, destination));
```

Comment:
I think this is a bug and ought to be
```suggestion
                        "%s = %s ? (%s %% %s): %s ;", destination, source, destination, source, source));
```
Same with line 564

Reply:
Assume destination = r1 and source = r2.

The instruction was:
r1 %= r2

After substitution

Current change:
r1 = r2 ? (r1 % r2) : r1;

Proposed change:
r1 = r2 ? (r1 % r2) : r2;

If (r2 == 0) then this becomes (r1 = 0) which doesn't match what I thought the required behavior was.

Reply:
Assume destination = r1 and source = r2.

The instruction was:
r1 %= r2

After substitution

Current change:
r1 = r2 ? (r1 % r2) : r1;

Proposed change:
r1 = r2 ? (r1 % r2) : r2;

If (r2 == 0) then this becomes (r1 = 0) which doesn't match what I thought the required behavior was.

Reply:
My bad, the original code was right and my doc was wrong.  It should be set to dest (r1) as current